### DISCUSSION TITLE
importing atom state into astro layout/page

### DISCUSSION THREAD

**Discussion by gplusdotgr** [ai/nanostores#195] [General]
```js
import { atom, action } from "nanostores";

export const brand = atom("wlw");

export const setBrand = action(brand, "setBrand", (store, brand) => {
  store.set(brand);
});
```
and the template
```html
<html lang="en" data-brand={brand.value}>
```

Why do I need to access it with `.value` ?

**Comment by ai** [comment]
Using `.value` directly is not recommended. 

What framework library do you for template?

**Reply by gplusdotgr** [reply]
Astro with vue, but this is a `.astro` layout/page

**Reply by ai** [reply]
According to https://github.com/nanostores/vue you should not use `.value`

**Reply by gplusdotgr** [reply]
sorry, I think I wasn't clear. This is an .astro file, not a vue component.  https://docs.astro.build/en/core-concepts/sharing-state/#why-nano-stores

**Reply by ai** [reply]
You need to subscribe to stores to use it (just calling `.value` doesn’t do subscriptions).

Vue and many other integrations have some functions like `useStore` which subscribe to store and return value. This is how you should work with Nano Stores.

On https://docs.astro.build/en/core-concepts/sharing-state/#why-nano-stores in a Vue tab I see a right way to work with stores. Where did you find a way to use `.value`?

**Reply by gplusdotgr** [reply]
I think we may not need nanostores for this functionality.. this is an astro middleware:
```js
import { setBrand } from "@store/brand";

export function onRequest(context, next) {
  console.log(context.request.url);
  if (context.request.includes === "europages") {
    setBrand("ep");
  } else {
    setBrand("wlw");
  }

  return next();
}
```

**Reply by ai** [reply]
Maybe. Sorry, I have no context of your application and environment.

**Reply by gplusdotgr** [reply]
I think one question you could help us by answered is should we use nanostores for a purely server-side rendered html . imagine: no client javascript.
We are getting the request url server-side and modify the HTML server-side.

**Reply by ai** [reply]
Hard to answer. Depends on architecture.

You should use Nano Stores when you need reactive updates. If there is something which should be updated on store update, it is a good place to use Nano Stores.

**Reply by gplusdotgr** [reply]
In our case there's no 'update' that will happen, the only way the value of the atom will change is a new request for the entire application/page. So I guess we don't want it to be reactive.

**Comment by gplusdotgr** [comment]
If someone else has feedback on this feel free to chime in.

**Comment by markpinero** [comment]
@gkatsanos you can set `context.locals.brand = x` in the middleware and set the value in the template `<html lang="en" data-brand={Astro.locals.brand}>`.

In any case, this use-case isn’t what nanostores is meant for if not reactive.

**Reply by gplusdotgr** [reply]
@markpinero indeed, we ended up using `locals`. 

We have a similar use case in which we want to set a nanostore value based on whether the user has an authorization cookie in his session. Would this be a usecase for nanostores?

**Reply by gplusdotgr** [reply]
(the value is not changed often/but we do need to check it often from different components of the application)
